android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java

```diff
@@ -200,6 +237,87 @@ public void checkDefaultPhoneAccount(Promise promise) {
         promise.resolve(telecomManager.getDefaultOutgoingPhoneAccount("tel") != null);
     }
 
+    @ReactMethod
+    public void setOnHold(String uuid, boolean shouldHold) {
+        Connection conn = VoiceConnectionService.getConnection(uuid);
+        if (conn == null) {
+            return;
+        }
+
+        if (shouldHold == true) {
+            conn.onHold();
+        } else {
+            conn.onUnhold();
+        }
+    }
+
+    @ReactMethod
+    public void reportEndCallWithUUID(String uuid, int reason) {
+        if (!isConnectionServiceAvailable() || !hasPhoneAccount()) {
+            return;
+        }
+
+        VoiceConnection conn = (VoiceConnection) VoiceConnectionService.getConnection(uuid);
+        if (conn == null) {
+            return;
+        }
+        conn.reportDisconnect(reason);
+    }
+
+    @ReactMethod
+    public void rejectCall(String uuid) {
+        if (!isConnectionServiceAvailable() || !hasPhoneAccount()) {
+            return;
+        }
+
+        Connection conn = VoiceConnectionService.getConnection(uuid);
+        if (conn == null) {
+            return;
+        }
+
+        conn.onReject();
+    }
+
+    @ReactMethod
+    public void setMutedCall(String uuid, boolean shouldMute) {
```

Comment:
When calling this method. I don't see the `mute` button updated on the native UI.
Have you ever succeeded to change it ?

Reply:
Hi Emmanuel, the native UI should update to correspond to the correct mute state, and does on our project. Can you please confirm you have this basic flow for each of your callkeep functions? 
Call setMutedCall in RN app -> callkeep tells the native sdk to update audio state -> sends a didPerformSetMutedCallAction event to the RN app -> In event callback perform the actual mute action.

Reply:
Hi @bhuangy !
Yes I receive the `didPerformSetMutedCallAction`. You can check it in the [example branch](https://github.com/react-native-webrtc/react-native-callkeep/pull/77) I've created.

In this example, you can start a call and mute it. It's works well on iOS but not on Android (on my Samsung J3).

Tell me if you want more details.

Reply:
@manuquentin I see the issue you're describing in your example app. I'm looking at it today. Everything looks reasonable at a glance...

Reply:
Hello, what's the status of this PR? Thank you :)

Reply:
@manuquentin I'm stumped on this. Your code looks right enough, but it's definitely not matching up properly. I've been pulled into other things, so I'm not actively looking at this right this second.

Reply:
@sboily Overall, I still think the PR is a serious improvement to the previous CallKeep release. @manuquentin has pointed out one place where he's seeing an issue that I've not been able to resolve yet and @danjenkins pointed out two small changes that should be addressed today or tomorrow by us. 

I think the real decision is "is this better", not "is this perfect", but I'm not the maintainer, I'm the one asking for the merge, ha. If we can get to a point where everyone is happy enough to merge, we can continue pushing additional, smaller, fixes upstream during development or join as contributors (your choice) to keep pushing this forward.

Reply:
Hello @kylekurz, this is my previous discussion with @manuquentin and that exactly my point. I'm agree, i think we need to merge to master, give example (#77) for people to test this new feature, and release a new version on few weeks. It give chance to another people to contribute on small bug instead a big branch. So i there is no objection, i'm agree to merge it. Thank you for your contribution to enhance callkeep :)

Reply:
I think we should wait until @bhuangy fixes Dan's issues (he's working on that now, should be done today) before merging.